addons/binding/org.openhab.binding.dlinksmarthome/src/main/java/org/openhab/binding/dlinksmarthome/internal/motionsensor/DLinkMotionSensorCommunication.java

```diff
@@ -0,0 +1,434 @@
+     * authentication data has been set. Device needs rebooting as it eventually becomes
+     * unresponsive.
+     *
+     * @throws SOAPException
+     */
+    private void buildRebootAction() throws SOAPException {
+        rebootAction.getSOAPHeader().detachNode();
+        final SOAPBody soapBody = rebootAction.getSOAPBody();
+        soapBody.addChildElement("Reboot", "", HNAP_XMLNS);
+
+        final MimeHeaders headers = rebootAction.getMimeHeaders();
+        headers.addHeader(SOAPACTION, REBOOT_ACTION);
+    }
+
+    /**
+     * Get the number of milliseconds to the next reboot time
+     *
+     * @return Time in ms to next reboot
+     */
+    private long getNextRebootTime() {
+        final LocalDateTime now = LocalDateTime.now();
+        final LocalDateTime todayStart = LocalDateTime.of(now.getYear(), now.getMonth(), now.getDayOfMonth(),
+                rebootHour, rebootMinute, 0);
+
+        LocalDateTime next = todayStart;
+        LocalDateTime nextStart = todayStart;
+
+        if (now.isBefore(todayStart)) {
+            final LocalDateTime yesterday = now.minusDays(1);
+            // Start from yesterday's reboot time
+            next = LocalDateTime.of(yesterday.getYear(), yesterday.getMonth(), yesterday.getDayOfMonth(), rebootHour,
+                    rebootMinute, 0);
+        } else {
+            final LocalDateTime tomorrow = now.plusDays(1);
+            // End at tomorrow's reboot time
+            nextStart = LocalDateTime.of(tomorrow.getYear(), tomorrow.getMonth(), tomorrow.getDayOfMonth(), rebootHour,
+                    rebootMinute, 0);
+        }
+
+        // Add intervals until we are past the current time
+        while (next.isBefore(now)) {
+            next = next.plusHours(rebootInterval);
+        }
+
+        // Each day starts at the reboot time
+        if (next.isAfter(nextStart)) {
+            next = nextStart;
+        }
+
+        return now.until(next, ChronoUnit.MILLIS);
+    }
+
+    /**
+     * Output unexpected responses to the debug log and sets the FIRMWARE error.
+     *
+     * @param message
+     * @param soapResponse
+     */
+    private void unexpectedResult(final String message, final Document soapResponse) {
+        logUnexpectedResult(message, soapResponse);
+
+        // Best guess when receiving unexpected responses
+        status = DeviceStatus.UNSUPPORTED_FIRMWARE;
+    }
+
+    /**
+     * Sends the two login messages and sets the authentication header for the action
+     * message.
+     *
+     * @param action
+     * @param timeout
+     */
+    private void login(final SOAPMessage action, final int timeout) {
+        loginSuccess = false;
+
+        login(timeout);
+        setAuthenticationHeaders(action);
+
+        switch (getHNAPStatus()) {
+            case LOGGED_IN:
+                loginSuccess = true;
+                break;
+            case COMMUNICATION_ERROR:
+                status = DeviceStatus.COMMUNICATION_ERROR;
```

Comment:
You are correct in that the states are similar, but not exactly the same, which is a result of the login side of the HNAP interface being pulled out into a base class for use by other devices. The motion sensor device has more states.